libc/CMakeLists.txt

```diff
@@ -52,7 +52,9 @@ set(PLATFORM_SRC "")
 if (OE_SGX)
   list(APPEND PLATFORM_SRC sgx/arc4random.c)
 else ()
-  list(APPEND PLATFORM_SRC
+  list(
```

Comment:
To follow-up on the previous thread on [OpenSSL oelibc dependnecies](https://github.com/openenclave/openenclave/pull/3335#discussion_r463862333):

* It's great that that we can eliminate most of the new dependencies in the oelibc API surface.
* Do we understand the impact of not providing link stubs for those previously added functions?
    * i.e. We will need to document the supported OpenSSL API surface in OE as we've done for other 3rdparty runtime libs, so it would be useful if knowing the missing symbols helps us backtrace unsupported OpenSSL functionality, or even more importantly, OpenSSL APIs that are supported but have different or limited behaviors.
* As an additional concern for the next PR, is there a way we can provide guidance for developers who end up in this failure case (e.g. they link some OpenSSL functionality with a missing underlying link symbol), which is similar to our discussions around the EDL opt-in missing syscalls experience. For example, have we thought about adding deprecation headers for unsupported functions when running in OE?

Reply:
The extra libc dependencies we previously had come from the following factors
-  We were building the OpenSSL separately (i.e., directly using the make-based build system from OpenSSL) and then linking against the libraries directly. This results in building the libraries with different compilation flags such that some extra APIs are linked. The dependencies fall in this category including `secure_getenv`, `h_errno`, `__xstat`, `pthread_cleanup_push`, and `pthread_setcancelstate`.
- We previously didn't disable `SECURE_MEM`. This introduces the dependencies of `madvise`, `mlock`, and `mprotect`. I added comments in the CMakeLists.txt.
- We previously didn't disable the `OPENSSL_UI_CONSOLE`, which introduces a set of OpenSSL APIs used by the OpenSSL command line tool. This introduces the dependencies of `tcgetattr` and `tcsetattr`.  I added comments in the CMakeLists.txt.

Per the investigation, the impact of eliminating those dependencies should be minimal. I'll go ahead and create document on supported OpenSSL APIs.